File: 2021-09/최대공약수최소공배수.js
function solution(n, m) {
    const gcd = getGreatestCommonFactor(n, m);
    return [gcd, getLeastCommonMultiple(n, m, gcd)];
}

function getGreatestCommonFactor(n, m) {
    const remainder = Math.floor(m % n);
    return remainder === 0 ? n : getGreatestCommonFactor(remainder, n);
}

function getLeastCommonMultiple(n, m, gcd) {
    return (n * m) / gcd;
}

console.log(solution(2, 5));

File: programmers/src/main/java/level1/최대공약수최소공배수.java
package level1;

public class 최대공약수최소공배수 {

	public int[] solution(int n, int m) {
		int min = Math.min(n, m);
		int gcd = 0;
		for (int i = 1; i <= min; i++) {
			if (n % i == 0 && m % i == 0)
				gcd = i;
		}

		return new int[]{gcd, (n * m) / gcd};
	}

}
